python/ray/_private/runtime_env/image_uri.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import os
+from typing import List, Optional
+
+from ray._private.runtime_env.context import RuntimeEnvContext
+from ray._private.runtime_env.plugin import RuntimeEnvPlugin
+
+default_logger = logging.getLogger(__name__)
+
+
+def _modify_context_impl(
+    image_uri: str,
+    run_options: Optional[List[str]],
+    worker_path: Optional[str],
+    context: RuntimeEnvContext,
+    logger: logging.Logger,
+    ray_tmp_dir: str,
+):
+    if worker_path:
+        context.override_worker_entrypoint = worker_path
+
+    container_driver = "podman"
+    container_command = [
+        container_driver,
+        "run",
+        "-v",
+        ray_tmp_dir + ":" + ray_tmp_dir,
+        "--cgroup-manager=cgroupfs",
+        "--network=host",
+        "--pid=host",
+        "--ipc=host",
+        # NOTE(zcin): Mounted volumes in rootless containers are
+        # owned by the user `root`. The user on host (which will
+        # usually be `ray` if this is being run in a ray docker
+        # image) who started the container is mapped using user
+        # namespaces to the user `root` in a rootless container. In
+        # order for the Ray Python worker to access the mounted ray
+        # tmp dir, we need to use keep-id mode which maps the user
+        # as itself (instead of as `root`) into the container.
+        # https://www.redhat.com/sysadmin/rootless-podman-user-namespace-modes
+        "--userns=keep-id",
+    ]
+
+    # The RAY_RAYLET_PID and RAY_JOB_ID environment variables are
+    # needed for the default worker.
+    container_command.append("--env")
+    container_command.append("RAY_RAYLET_PID=" + os.getenv("RAY_RAYLET_PID"))
+    container_command.append("--env")
+    container_command.append("RAY_JOB_ID=$RAY_JOB_ID")
+    for env_var_name, env_var_value in os.environ.items():
```

Comment:
overall lgtm, just 1 question: does this image_uri really work with env_vars?

so setup_worker.py sets the env vars to itself's process and then execvp the py_executable with args. Now, py_executable is set to podman run but it does not (normally?) inherit env vars from the host process. 

Here in image_uri.py we log current proc's env vars to the podman command, but this should be the env vars from *the runtime env agent process* not the ones we wanted from *the target worker*. To mitigate we updated RAY_RAYLET_PID and RAY_JOB_ID but not any user `env_vars`.

This worked for `container` but not for `image_uri` because the former declared it does not support env vars, but the latter supports.

I think the proper fix is, since we can change `py_exectuable`, is to read all envs on-the-fly and pass them to podman like this:

` --env-file <(env) `

to become

```
podman run -v /tmp/ray:/tmp/ray \
    --cgroup-manager=cgroupfs --network=host --pid=host --ipc=host \
    --userns=keep-id --env-file <(env) \
    --entrypoint python rayproject/ray:nightly-py39
```

And in code, remove all these --env and add

```
container_command.append("--env-file")
container_command.append("<(env)")
```

Reply:
One thing to think about is if this is compatible with `container` which used to not inherit user env vars. For example we can write 2 funcs to treat `container` the old way and document it well, and `image_uri` the new way.